common/debug-interface/src/libra_trace.rs

```diff
@@ -296,3 +333,60 @@ fn abbreviate_crate(name: &str) -> &str {
         _ => name,
     }
 }
+
+// This is exact copy of similar function in log crate
+/// Sets libra trace config
+pub fn set_libra_trace(config: &HashMap<String, String>) -> Result<(), ()> {
+    unsafe {
+        match LIBRA_TRACE_STATE.compare_and_swap(UNINITIALIZED, INITIALIZING, Ordering::SeqCst) {
+            UNINITIALIZED => {
+                SAMPLING_CONFIG = Some(parse_sampling_config(config));
+                LIBRA_TRACE_STATE.store(INITIALIZED, Ordering::SeqCst);
+                Ok(())
+            }
+            INITIALIZING => {
+                while LIBRA_TRACE_STATE.load(Ordering::SeqCst) == INITIALIZING {}
+                Err(())
+            }
+            _ => Err(()),
+        }
+    }
+}
+
+fn parse_sampling_config(config: &HashMap<String, String>) -> Sampling {
+    Sampling(
+        config
+            .iter()
+            .map(|(k, v)| {
+                let k: &'static str = Box::leak(k.clone().into_boxed_str());
+                let v = v.split('/').collect::<Vec<&str>>();
+                let v = CategorySampling {
+                    nominator: v[0].parse::<u64>().unwrap(),
+                    denominator: v[1].parse::<u64>().unwrap(),
+                };
+                (k, v)
+            })
+            .collect::<HashMap<&'static str, CategorySampling>>(),
+    )
+}
+
+/// Checks if libra trace is enabled
+pub fn libra_trace_set() -> bool {
+    LIBRA_TRACE_STATE.load(Ordering::SeqCst) == INITIALIZED
+}
+
+pub fn is_selected(node: (&str, &Vec<u8>)) -> bool {
+    unsafe {
```

Comment:
We should verify if LIBRA_TRACE_STATE is initialized before reading `&SAMPLING_CONFIG` it to avoid partial reads